builds/ci/mqtt.yaml

```diff
@@ -1,19 +1,127 @@
 jobs:
-  ################################################################################
-  - job: compliance_amd64
-    ################################################################################
-    displayName: Compliance amd64
+
+################################################################################
+  - job: check_run_pipeline
+################################################################################
+    displayName: Check pipeline preconditions (changes ARE in builds or mqtt)
     pool:
       vmImage: "ubuntu-16.04"
     steps:
-      - script: echo "##vso[task.setvariable variable=RUST_BACKTRACE;]1"
-        displayName: Set env variables
-      - bash: mqtt/build/linux/install.sh
+      - bash: |
+          git log -m -1 --name-only --first-parent --pretty="" | egrep -i '^(builds|mqtt)'
+          if [[ $? == 0 ]]; then
+            echo "Detected changes inside builds or mqtt folders"
+            echo "##vso[task.setvariable variable=RUN_PIPELINE;isOutput=true]TRUE"
+          fi
+        displayName: Check changes in sources
+        name: check_files
+
+################################################################################
+  - job: linux_amd64
+################################################################################
+    displayName: Linux amd64
+    timeoutInMinutes: 30
+    dependsOn: check_run_pipeline
+    condition: eq(dependencies.check_run_pipeline.outputs['check_files.RUN_PIPELINE'], 'true')
+    pool:
+      vmImage: 'ubuntu-16.04'
+    steps:
+      - task: Bash@3
+        displayName: Install Rust
+        inputs:
+          filePath: mqtt/build/linux/install.sh
+      - task: Bash@3
+        displayName: Build
+        inputs:
+          filePath: mqtt/build/linux/build.sh
+      - task: Bash@3
+        displayName: Test
+        inputs:
+          filePath: mqtt/build/linux/test.sh
+
+################################################################################
+  - job: linux_arm32v7
+################################################################################
+    displayName: Linux arm32v7
+    timeoutInMinutes: 30
+    dependsOn: check_run_pipeline
+    condition: eq(dependencies.check_run_pipeline.outputs['check_files.RUN_PIPELINE'], 'true')
+    pool:
+      vmImage: 'ubuntu-16.04'
+    steps:
+      - script: |
+          echo "##vso[task.setvariable variable=RUSTUP_HOME;]$(Agent.WorkFolder)/rustup"
+          echo "##vso[task.setvariable variable=CARGO_HOME;]$(Agent.WorkFolder)/cargo"
+          echo "##vso[task.setvariable variable=PATH;]$(Agent.WorkFolder)/cargo/bin:$PATH"
+        displayName: Modify path
+      - task: Bash@3
         displayName: Install Rust
-      - bash: mqtt/build/linux/build.sh -r 1
+        inputs:
+          filePath: mqtt/build/linux/install.sh
+      - script: cargo install cross --version 0.1.16
+        displayName: Install cross
+      - task: Bash@3
         displayName: Build
-      - bash: mqtt/build/linux/compliance.sh
```

Comment:
do we still need to run this script?

Reply:
Not anymore. all 3rd party tests were moved into rust.